samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,114 @@ def _validate_deployment_preference_and_add_update_policy(
+        "Runtime": PropertyType(True, is_str()),
+        "CodeUri": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "InlineCode": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "MemorySize": PropertyType(False, is_type(int)),
+        "Tags": PropertyType(False, is_type(dict)),
+        "Tracing": PropertyType(False, is_type(bool)),
+        "Timeout": PropertyType(False, is_type(int)),
+        "Role": PropertyType(True, is_str()),
+        "Schedule": PropertyType(True, one_of(is_type(dict), is_str())),
+        "StartCanaryAfterCreation": PropertyType(True, is_type(bool)),
+        "ArtifactS3Location": PropertyType(True, one_of(is_type(dict), is_str())),
+        "FailureRetentionPeriod": PropertyType(False, one_of(is_type(dict), is_str())),
+        "SuccessRetentionPeriod": PropertyType(False, one_of(is_type(dict), is_str())),
+        "VpcConfig": PropertyType(False, is_type(dict)),
+        "Environment": PropertyType(False, dict_of(is_str(), is_type(dict))),
+    }
+
+    def to_cloudformation(self, **kwargs):
+        """Returns the Synthetics Canary to which this SAM Canary corresponds.
+
+        :param dict kwargs: already-converted resources that may need to be modified when converting this \
+        macro to pure CloudFormation
+        :returns: a list of vanilla CloudFormation Resources, to which this Function expands
+        :rtype: list
+        """
+        resources = []
+        syntheticsCanary = self._construct_synthetics_canary()
+        resources.append(syntheticsCanary)
+
+        return resources
+
+    def _construct_synthetics_canary(self):
+        """Constructs a AWS::Synthetics::Canary resource."""
+        canary = SyntheticsCanary(
+            self.logical_id, depends_on=self.depends_on, attributes=self.get_passthrough_resource_attributes()
+        )
+        canary.ArtifactS3Location = self.ArtifactS3Location
+        canary.Code = self._construct_code_dict()
+        canary.ExecutionRoleArn = self.Role
+        canary.FailureRetentionPeriod = self.FailureRetentionPeriod
+        # Make a default for canary name
+        canary.Name = self.FunctionName
+        canary.RuntimeVersion = self.Runtime
+        canary.Schedule = self.Schedule
+        canary.StartCanaryAfterCreation = self.StartCanaryAfterCreation
+        canary.SuccessRetentionPeriod = self.SuccessRetentionPeriod
+        canary.Tags = self._construct_tag_list(self.Tags)
+        canary.VPCConfig = self.VpcConfig
+
+        if self.Tracing or self.Environment or self.MemorySize or self.Timeout:
+            canary.RunConfig = self._construct_run_config()
+        return canary
+
+    def _construct_run_config(self):
+        runconfig = {"ActiveTracing": self.Tracing, "MemoryInMB": self.MemorySize, "TimeoutInSeconds": self.Timeout}
+        if self.Environment:
+            runconfig["EnvironmentVariables"] = self.Environment["Variables"]
+
+        return {k: v for k, v in runconfig.items() if v is not None}
```

Comment:
It took me couple of seconds to understand that you are filtering out `None` properties. This might be a useful method, can you probably extract out this code to a separate method like:

```
def extract_not_none_properties(self, dict):
    return {k: v for k, v in dict.items() if v is not None}
```

this way somebody who is reading the code understands it instantly and you can also unit test this logic in more ways.